### DISCUSSION TITLE
Drop ES5 Javascript Files

### DISCUSSION THREAD

**Discussion by dgrammatiko** [joomla/joomla-cms#35737] [Q&A]
# Another request for the Production Team/Maintainers

### Current state
IE11 was never supposed to be supported in Joomla 4.x mainly because the decision to move to Bootstrap 5 made this impossible. At the time @wilsonge and I worked on the Bootstrap 5 I also did a bunch of PRs to introduce `type="modules"`/`nomodule` attributes that allowed the CMS to deliver modern Javascript but also retain the option to deliver ES5 (code compatible with IE11) as a solution to this discussion: https://github.com/joomla/joomla-cms/discussions/32148. The solution was a compromise for the time as the project wanted to provide working javascript files for anyone that wanted to build something that could support IE11. The problem is that the ES5 files were:
- never tested that actually worked in the IE11
- don't have all the required polyfills
- there isn't a way in the PHP part of the CMS to define and serve polyfills (debatable as the WAM has dependencies and could be somehow patched to do that, the point for polyfills is that they need to run before any other code and maybe even before CSS, depending what they're polyfilling, also need to be synchronous & render-blocking)

### Current WEB state
This year most of the most known WEB entities already publicly declared that they stop supporting IE11, starting with Microsoft itself and their flagship Office 365. Lately, Google made a similar announcement but, and this is the interesting part here, WP dropped support on May 2021: https://wordpress.org/news/2021/05/dropping-support-for-internet-explorer-11/

### The request
What I'm requesting here is a decision from the PT that either:
- Joomla will continue shipping ES5 javascript files (explaining the reasons)
- Joomla will stop shipping ES5 javascript files (for the reasons explained briefly above, eg everybody drops ES5)

If the decision is in favour of dropping the ES5 javascript files, there will be a few build tool files that will need some patching and then the `joomla.assets.json` files where we need to drop the dependencies to ES5 (the way we end up shipping `type=module/nomodule`. Also, the package will get thinner as a lot of files will be removed from the zip files.
This could be done only on a minor version, so basically the decision should be targeted for v4.1

**Comment by PhilETaylor** [comment]
s/Amother/Another

Possibly better placed in https://github.com/joomla/rfc ?

**Reply by dgrammatiko** [reply]
Here will get more attention. Although this is a direct question towards the PT/Maintainers I choose to do that publically so in case anyone disagrees could speak their minds. Actually this is the continuation of the https://github.com/joomla/joomla-cms/discussions/32148 discussion that we end up monkey patching at that time because the product had to ship at some point. Now that the time pressure is not here we could revisit the decisions and the solutions and make any adjustments if needed.

Anyways, if there's a new guideline that everything should be done through this new RFC I'll close this

**Comment by krileon** [comment]
I still stand by my original discussion that ES5 should be dropped.

I don't think it's appropriate for Joomla 4 to output ES5 **but not test it**. Joomla 4 either supports ES5 or it doesn't. Pick 1. IE11 is dead. Microsoft doesn't support it and now Google search doesn't support it. WP also doesn't support it. Joomla 4 should just output ES6.

**Reply by dgrammatiko** [reply]
FWIW there's a PR https://github.com/joomla/joomla-cms/pull/35739 but ultimately it's all about the decision to drop them or not

**Comment by laoneo** [comment]
I guess we have to decide which minimum Safari browser we do support. If we drop, then iOS versions < 11 will also not be supported anymore.
